src/hkdfchachapoly.rs

```diff
@@ -0,0 +1,150 @@
+use ring::aead;
+use byteorder::{ByteOrder, BigEndian, LittleEndian};
+use hkdf::Hkdf;
+use sha2::Sha256;
+use error::Error;
+
+pub struct HkdfChaChaPoly {
+    key: [u8; 32]
```

Comment:
Might want to use the `clear_on_drop` crate on this value to ensure this is wiped on `Drop`.

(I tend to just `impl Drop` and use the `clear_on_drop::clear::Clear` trait to clear the value, to avoid dealing with the `Borrow`/`Deref`-based API to `ClearOnDrop`

Reply:
Will do!